### PR TITLE
Add Centroid in `window_dict` output

### PR DESCRIPTION
**Pull request by KrishPatel13**
**What kind of change does this PR introduce?**

**Summary**

**Checklist**
* [x] My code follows the style guidelines of OpenAdapt
* [x] I have performed a self-review of my code
* [ ] If applicable, I have added tests to prove my fix is functional/effective
* [x] I have linted my code locally prior to submission
* [x] I have commented my code, particularly in hard-to-understand areas
* [ ] I have made corresponding changes to the documentation (e.g. README.md, requirements.txt)
* [x] New and existing unit tests pass locally with my changes

**How can your code be run and tested?**

**Other information**

### CODE REVIEW

**Comment by KrishPatel13** [comment]
@abrichr Any suggestion on how to test this PR practically? 
I have tried theoretically here:

**Comment by KrishPatel13** [comment]
Ready for review @abrichr !

File: openadapt/models.py [hunk 1]
```diff
@@ -572,6 +572,31 @@ def to_prompt_dict(self, include_data: bool = True) -> dict[str, Any]:
         else:
             window_dict["state"].pop("data")
         window_dict["state"].pop("meta")
+
+        # Let,
+        # left = window_dict["left"] = x
+        # top = window_dict["top"] = y
+        # width = window_dict["width"] = w
+        # height = window_dict["height"] = h
```
Comment (abrichr) [inline openadapt/models.py:580]: What do you think about uncommenting these and using them in the calculation below?

File: openadapt/models.py [hunk 2]
```diff
@@ -572,6 +572,31 @@ def to_prompt_dict(self, include_data: bool = True) -> dict[str, Any]:
...
+
+        # Compute the centroid of the bounding box
+        # Reference: https://stackoverflow.com/a/18932029
```
Comment (abrichr) [inline openadapt/models.py:583]: I don't think we need a reference here, this is self explanatory

File: openadapt/models.py [hunk 3]
```diff
@@ -572,6 +572,31 @@ def to_prompt_dict(self, include_data: bool = True) -> dict[str, Any]:
...
+            "x": centroid_x,
+            "y": centroid_y,
+        }
+
+        # Uncomment, if we want to remove the x, y, w, h data,
+        # if using centroid solves all of the issues.
+        # window_dict.pop("left")
+        # window_dict.pop("top")
+        # window_dict.pop("width")
+        # window_dict.pop("height")
```
Comment (abrichr) [inline openadapt/models.py:598]: What do you think about adding an argument to control whether this happens or not, e.g. `add_centorid=True, remove_bbox=False`?

**Review by abrichr — CHANGES_REQUESTED** [review]
Thank you @KrishPatel13 ! Just a few small comments.

File: openadapt/models.py (hunk 1 above)
Comment (KrishPatel13) [reply openadapt/models.py]: Done here: 5d4bb1c24311d0c6fe387a7b6143342288048cf7

File: openadapt/models.py (hunk 2 above)
Comment (KrishPatel13) [reply openadapt/models.py]: Done here: 5d4bb1c24311d0c6fe387a7b6143342288048cf7

File: openadapt/models.py (hunk 3 above)
Comment (KrishPatel13) [reply openadapt/models.py]: Done here: 5d4bb1c24311d0c6fe387a7b6143342288048cf7

**Comment by KrishPatel13** [comment]
@abrichr Ready for Review!

**Pull request merged**